strictyaml/yaml.go

```diff
@@ -0,0 +1,28 @@
+// Package strictyaml provides a strict YAML unmarshaller based on `go-yaml/yaml`
+package strictyaml
+
+import (
+	"bytes"
+	"errors"
+	"io"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Unmarshal takes a byte array and an arbitrary interface as arguments and
+// attempts to unmarshal the contents of the byte array into a defined struct. Any
+// config keys from the incoming YAML document which do not correspond to
+// expected keys in the config struct will result in errors.
```

Comment:
I would reword this to be closer to the docs for `Decoder.Decode()`.

> Decode reads the next YAML-encoded value from its input and stores it in the value pointed to by v.

We want callers of this function to know that they need to pass the holder `yamlObj` by reference.